Assets/Scripts/PuzzleScript/PuzzleManager.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PuzzleManager : MonoBehaviour
+{
+    public static PuzzleManager Instance; 
+
+    public TriggerLogger[] triggerLogger; 
+
+    private void Awake()
+    {
+        if (Instance == null)
+        {
+            Instance = this;
+        }
+        else
+        {
+            Destroy(gameObject);
+        }
+    }
+
+    public void CheckPuzzleCompletion()
+    {
+        foreach (TriggerLogger cruz in triggerLogger)
```

Comment:
Could you explain why the variable “cruz” is in Spanish, it would be better if all the content of the code was in English.